.github/workflows/cleanup_cache.yml

```diff
@@ -0,0 +1,63 @@
+name: Cleanup cache
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: write
+      contents: read
+
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      REPO: ${{ github.repository }}
+      BRANCH: refs/pull/${{ github.event.pull_request.number }}/merge
+
+    steps:
+      - name: Preconditions (Install gh-actions-cache)
+        run: |
+          gh extension install actions/gh-actions-cache
+
+      - name: Cleanup PR's caches
+        run: |
+          echo "Fetching list of cache keys for ${BRANCH}"
+          CACHE_KEYS_FOR_PR=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+
+          set +e
+          echo "Deleting caches..."
+          for CACHE_KEY in $CACHE_KEYS_FOR_PR
+          do
+            gh actions-cache delete $CACHE_KEY -R $REPO -B $BRANCH --confirm
+          done
+          echo "Done"
+
+      - name: Cleanup master's caches
+        if: github.event.pull_request.merged == true
+        env:
+          BRANCH: master
+          KEYS: "ccache-linux-ubuntu ccache-Build-linux-debian ccache-Build-linux-android ccache-Build-linux-arm64"
```

Comment:
Do they find the cache entries with timestamps in them?

Reply:
yes, it finds all entries including entries with timestamps